File: TiketVelvet.java
public class TiketVelvet extends Tiket {
    // Properti tambahan khusus tiket Velvet
    private boolean bantalSelimutPack;
    private boolean layananButler;

    // Constructor
    public TiketVelvet(int idTiket, String namaFilm, String jadwalTayang, int jumlahKursi, int hargaDasarTiket, boolean bantalSelimutPack, boolean layananButler) {
        super(idTiket, namaFilm, jadwalTayang, jumlahKursi, hargaDasarTiket);
        this.bantalSelimutPack = bantalSelimutPack;
        this.layananButler = layananButler;
    }

    @Override
    public double hitungTotalHarga() {
        // Hint: Kalau mengacu ke contoh soal dosenmu di Tahap 2, 
        // overriding harga Velvet bakal dilakukan di Tahap 5.
        // Jadi sementara biarkan return harga dasar dulu.
        return hargaDasarTiket;
    }

    @Override
    public void tampilkanInfoFasilitas() {
        System.out.println("--- Fasilitas Studio Velvet ---");
        System.out.println("Bantal & Selimut : " + (bantalSelimutPack ? "Disediakan" : "Tidak Disediakan"));
        System.out.println("Layanan Butler   : " + (layananButler ? "Tersedia" : "Tidak Tersedia"));
    }
}

File: TiketVelvet.php
<?php
require_once 'Tiket.php';

class TiketVelvet extends Tiket {
    private $bantal_selimut_pack;
    private $layanan_butler;

    public function __construct($id_tiket, $nama_film, $jadwal_tayang, $jumlah_kursi, $harga_dasar_tiket, $bantal_selimut_pack, $layanan_butler) {
        parent::__construct($id_tiket, $nama_film, $jadwal_tayang, $jumlah_kursi, $harga_dasar_tiket);
        $this->bantal_selimut_pack = $bantal_selimut_pack;
        $this->layanan_butler = $layanan_butler;
    }

    // Overriding Tahap 5
    public function hitungTotalHarga() {
        return ($this->jumlah_kursi * $this->harga_dasar_tiket) * 1.50;
    }

    public function tampilkanInfoFasilitas() {
        $selimut = $this->bantal_selimut_pack ? "Ya" : "Tidak";
        $butler = $this->layanan_butler ? "Ya" : "Tidak";
        return "Bantal/Selimut: {$selimut} | Butler: {$butler}";
    }
}
?>
